### PR TITLE
[CSL-1181] Remove delegation from poor guys

### PR DESCRIPTION
**Pull request by gromakovsky**

### CODE REVIEW

**Comment by gromakovsky** [comment]
@volhovM I've just added rollback logic. Please review my last commit carefully.

File: src/Pos/Delegation/Logic/VAR.hs [hunk 1]
```diff
@@ -341,7 +361,11 @@ dlgVerifyBlocks blocks = do
         HashSet StakeholderId ->
         Block ssc ->
         ExceptT Text (MapCede (StateT DlgVerState m)) DlgUndo
-    verifyBlock _ (Left _) = [] <$ (dvCurEpoch .= HS.empty)
+    verifyBlock _ (Left genesisBlk) = do
+        dvCurEpoch .= HS.empty
+        let blkEpoch = genesisBlk ^. epochIndexL
+        noLongerRichmen <- getNoLongerRichmen blkEpoch
+        catMaybes <$> mapM getPsk noLongerRichmen
```
Comment (volhovM) [inline src/Pos/Delegation/Logic/VAR.hs]: I think you also need to remove it from `Cede` context (`modPsk`). Or not. Depending on whether we implement "don't verify over genesis block" policy. If we do, you should.

File: src/Pos/Delegation/Logic/VAR.hs [hunk 2]
```diff
@@ -512,15 +536,10 @@ removeNoLongerRichmen ::
...
-    let getRichmen e =
-            toList <$>
-            lrcActionOnEpochReason e "removeNoLongerRichmen" LrcDB.getRichmenDlg
-    oldRichmen <- getRichmen (newEpoch - 1)
-    newRichmen <- getRichmen newEpoch
-    let edgeActions = map DlgEdgeDel $ oldRichmen \\ newRichmen
-    calculateTransCorrections $ HS.fromList edgeActions
+    noLongerRichmen <- getNoLongerRichmen newEpoch
+    let edgeActions = map DlgEdgeDel noLongerRichmen
+    calculateTransCorrections (HS.fromList edgeActions)
```
Comment (volhovM) [inline src/Pos/Delegation/Logic/VAR.hs]: This is weird. You should return edge actions with trans corrections. Edge actions modify db (1), while trans corrections affect transitive and transitive reverse (2-3).

File: src/Pos/Delegation/Logic/VAR.hs (hunk 1 above)
Comment (gromakovsky) [inline src/Pos/Delegation/Logic/VAR.hs]: It seems easy to do, so I will do it even if it's not necessary (I don't want to rely on it when it's avoidable).

**Comment by gromakovsky** [comment]
@volhovM I've just fixed everything.

**Comment by gromakovsky** [comment]
@volhovM do you think you will finish CSL-1205 today? If yes, I won't merge this PR before #1032.

**Pull request merged**